src/vcf/fixer.cpp

```diff
@@ -61,106 +67,135 @@ namespace ebi
             const size_t info_column_index = 7;
-            const std::string empty_info_column = MISSING_VALUE;
-
-            size_t num_removed_fields = 0;
             std::string string_line = {line->begin(), line->end()};
 
-            auto condition_to_remove_info_field = [&](std::string &info_subfield, size_t index) -> bool {
-                std::vector<std::string> key_value;
-                util::string_split(info_subfield, "=", key_value);
-                return key_value[0] == error.field;
-            };
-
             fix_column(info_column_index, string_line, "\t", [&](std::string &info_column) {
-                num_removed_fields = remove_column(info_column, ";", empty_info_column, condition_to_remove_info_field);
+                const std::string empty_info_column = MISSING_VALUE;
+
+                auto condition_to_modify_info_field = [&](const std::string &info_subfield, size_t index) -> bool {
+                    std::vector<std::string> key_value;
+                    util::string_split(info_subfield, "=", key_value);
+                    return key_value[0] == error.field;
+                };
+
+                if (error.error_fix == ErrorFix::DUPLICATE_VALUES) {
+                    std::cerr << "DEBUG: line " << error.line << ": fixing duplicate INFO fields" << std::endl;
+                    remove_duplicate_key_value_pairs(info_column, ";", "=", empty_info_column);
+                } else if (error.error_fix == ErrorFix::RECOVERABLE_VALUE) {
+                    std::cerr << "DEBUG: line " << error.line << ": fixing invalid INFO field " << error.field << std::endl;
+                    replace_fields(info_column, ";", error.field + "=" + error.expected_value, condition_to_modify_info_field);
+                } else if (error.error_fix == ErrorFix::IRRECOVERABLE_VALUE) {
+                    std::cerr << "DEBUG: line " << error.line << ": removing invalid INFO field " << error.field << std::endl;
+                    remove_fields(info_column, ";", empty_info_column, condition_to_modify_info_field);
+                }
             });
-
-            if (num_removed_fields != 1) {
-                std::cerr << "WARNING: line " << error.line << ": field " << error.field << " appeared "
-                          << num_removed_fields << " times " << std::endl;
-            }
         }
 
         void Fixer::visit(FormatBodyError &error)
         {
-            util::writeline(output, *line);
-            ignored_errors++;
+            if (error.error_fix != ErrorFix::DUPLICATE_VALUES) {
+                ignore_error();
+                return;
+            }
+        
+            std::cerr << "DEBUG: line " << error.line << ": fixing duplicate FORMAT fields" << std::endl;
+
+            std::string string_line = {line->begin(), line->end()};
+
+            remove_duplicate_format_sample_pairs(string_line);
```

Comment:
It would be more clear to wrap this block inside an `if (error.error_fix == ErrorFix::DUPLICATE_VALUES)`, and calling `ignore_error` otherwise. The current way would require more complex code changes if more ErrorFix are supported in the future.